### ISSUE TITLE
Incorrect indentation in generated file

### ISSUE THREAD

**Issue by trosh** [xflr6/graphviz#43]
This is all tested with the graphviz python package in version `0.7.1`, which is currently the latest.

Here is a test script

```py
#!/usr/bin/env python2

import graphviz

graph = graphviz.Graph()
graph.edge("a", "b")
subgraph = graphviz.Graph(body=["rank=same"])
subgraph.node("sub", label="hero")
subgraph.edge("sub", "zero")
graph.subgraph(subgraph)
graph.edge("a", "sub")
graph.edge("b", "zero")

print graph
```

which generates the following output

```
graph {
                a -- b
        {
        rank=same
                sub [label=hero]
                        sub -- zero
        }
                a -- sub
                b -- zero
}
```

We can see that some lines are incorrectly indented: in this case the edge lines and the rank line.

What we expect in this case is the following

```
graph {
        a -- b
        {
                rank=same
                sub [label=hero]
                sub -- zero
        }
        a -- sub
        b -- zero
}
```

Here is the graph, as rendered by dot, for completeness:

![graphvizbadindent](https://cloud.githubusercontent.com/assets/3617724/26636941/6cc31e50-461e-11e7-8db0-57705d428843.png)

**Comment by xflr6** [comment]
The additional indentation for edge statments is (at least currently) intended (see the [examples in the docs](https://graphviz.readthedocs.io/en/latest/manual.html)).

In principle, you can override it as a quick hack in a subclass by dropping the second `\t` in its `_edge_` and `_edge_plain` attributes [here](https://github.com/xflr6/graphviz/blob/d506110f062f5344af197e490eef979c713dfc07/graphviz/dot.py#L249-L250) (or directly change it on `Graph` and `Digraph`), but as these are private attributes, there are no guarantees that this will continue to work.

As [shown in the docs](https://graphviz.readthedocs.io/en/stable/manual.html#custom-dot-statements) the library will do nothing to the strings you put into (or append to) `body`. If you use `subgraph.attr(rank='same')` instead of adding custom DOT, the indentation of the rank statement should be on the same level as the nodes.

**Comment by trosh** [comment]
Regarding extra indentation for edges: oops, had missed that in the doc. Now this gets into the realm of aesthetics in which debate is difficult, but since an edge statement can implicitely declare node(s) in DOT language (as is common usage), it seems strange to me to create such a visual discrepancy between these statements. (The DOT examples in the graphviz dotguide don't do this &mdash; that's not necessarily a good argument but I would argue they are more readable.) At least a parameter to Graph/Digraph explicitely controlling that behaviour would seem an improvement.

Regarding custom DOT indentation, thanks for explaining what was going on; I'll change to an `attr` statement as is indeed explained in the doc. However I feel that adding an extra level of indentation to custom DOT would make sense: why would any graph's body not be indented?

Thanks for the good and fast answer.

**Comment by xflr6** [comment]
Thanks, will consider changing the default or adding configurability (not sure though how much users generally care about the aethetics of the generated source, so maybe no high priority for now).

You can also use `subgraph()` as a context manager in your example:
```python
with graph.subgraph() as subgraph:
    subgraph.node("sub", label="hero")
    subgraph.edge("sub", "zero")
```

**Comment by trosh** [comment]
I do agree that generated sources are probably transparent to the user and therefore low priority. Just the OCD talking here 😃 

The `with as` syntax is cute !

**Comment by xflr6** [comment]
fixed in 22a7ecb117abb66cabc0aab4805eaad4a1587e0a
